File: project-tab/adapter-shim/openai/adapter_shim/brief_to_prompt.py
"""Convert an AgentBrief into a prompt string for the Codex CLI."""

from __future__ import annotations

from .models import AgentBrief


def brief_to_prompt(brief: AgentBrief, *, continuation: bool = False) -> str:
    """Render an AgentBrief as a structured prompt string for codex exec."""
    sections: list[str] = []

    if continuation:
        sections.append(
            "Your previous assignment is complete. Here is your next assignment:\n"
        )

    sections.append(f'You are a {brief.role} working on the "{brief.workstream}" workstream.')
    sections.append(brief.description)

    # Project
    pb = brief.project_brief
    sections.append(f"\n## Project\n{pb.title}: {pb.description}")

    # Goals
    if pb.goals:
        goals = "\n".join(f"- {g}" for g in pb.goals)
        sections.append(f"\n## Goals\n{goals}")

    # Constraints
    constraints = brief.constraints or []
    if pb.constraints:
        constraints = constraints + pb.constraints
    if constraints:
        cons = "\n".join(f"- {c}" for c in constraints)
        sections.append(f"\n## Constraints\n{cons}")

    # Knowledge snapshot summary
    ks = brief.knowledge_snapshot
    if ks and ks.estimated_tokens > 0:
        parts: list[str] = []
        if ks.workstreams:
            parts.append(f"{len(ks.workstreams)} active workstream(s)")
        if ks.pending_decisions:
            parts.append(f"{len(ks.pending_decisions)} pending decision(s)")
        if ks.artifact_index:
            parts.append(f"{len(ks.artifact_index)} artifact(s)")
        if parts:
            sections.append(f"\n## Context\n{', '.join(parts)}.")

    result = "\n".join(sections)
    # Rough cap at ~2000 tokens (~8000 chars)
    if len(result) > 8000:
        result = result[:7997] + "..."
    return result

File: project-tab/adapter-shim/claude/src/brief-to-prompt.ts
/**
 * Convert an AgentBrief into a prompt string for the Claude CLI.
 */

import type { AgentBrief } from './models.js'

export function briefToPrompt(
  brief: AgentBrief,
  options?: { continuation?: boolean }
): string {
  const sections: string[] = []

  if (options?.continuation) {
    sections.push(
      'Your previous assignment is complete. Here is your next assignment:\n'
    )
  }

  sections.push(`You are a ${brief.role} working on the "${brief.workstream}" workstream.`)
  sections.push(brief.description)

  // Project
  const pb = brief.projectBrief
  sections.push(`\n## Project\n${pb.title}: ${pb.description}`)

  // Goals
  if (pb.goals?.length) {
    const goals = pb.goals.map(g => `- ${g}`).join('\n')
    sections.push(`\n## Goals\n${goals}`)
  }

  // Constraints
  const constraints = [...(brief.constraints ?? []), ...(pb.constraints ?? [])]
  if (constraints.length) {
    const cons = constraints.map(c => `- ${c}`).join('\n')
    sections.push(`\n## Constraints\n${cons}`)
  }

  // Knowledge snapshot summary
  const ks = brief.knowledgeSnapshot
  if (ks && ks.estimatedTokens > 0) {
    const parts: string[] = []
    if (ks.workstreams?.length) parts.push(`${ks.workstreams.length} active workstream(s)`)
    if (ks.pendingDecisions?.length) parts.push(`${ks.pendingDecisions.length} pending decision(s)`)
    if (ks.artifactIndex?.length) parts.push(`${ks.artifactIndex.length} artifact(s)`)
    if (parts.length) sections.push(`\n## Context\n${parts.join(', ')}.`)
  }

  let result = sections.join('\n')
  // Rough cap at ~2000 tokens (~8000 chars)
  if (result.length > 8000) {
    result = result.slice(0, 7997) + '...'
  }
  return result
}
